### DISCUSSION TITLE
Create lib to allow use in other types of code

### DISCUSSION THREAD

**Discussion by SombraRO** [zhad3/zrenderer#48] [Ideas]
Hello @zhad3, I found your project very interesting, so I wanted to bring this suggestion, to also implement a lib for use in other languages ​​such as Csharp/C/C++

**Comment by zhad3** [comment]
Hi @SombraRO , how would you use the library?

zrenderer wasn't designed to be used as a library. If its just about triggering the creation of sprites then you could probably just call the CLI tool?

**Comment by SombraRO** [comment]
The idea would be to integrate functions into a C++/C# code itself, not necessarily just saving the image. For example, rendering an image through the bytes generated in a WPF launcher or in a C++ code.

**Comment by zhad3** [comment]
Yes, I am aware how a library is used. I am mostly interested what functionality it should provide. Currently I do not have the motivation to think up an interface that could in theory be useful for (to me) unknown applications. It is easier to have a clear goal of the functionality it should provide.
